spring-cloud-task-stream/src/main/java/org/springframework/cloud/task/listener/TaskEventAutoConfiguration.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.task.listener;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.Output;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.gateway.GatewayProxyFactoryBean;
+import org.springframework.messaging.MessageChannel;
+
+/**
+ *
+ * @author Michael Minella
+ */
+@Configuration
+@ConditionalOnClass(EnableBinding.class)
+@ConditionalOnBean(TaskLifecycleListener.class)
+@ConditionalOnProperty(prefix = "spring.cloud.task.events", name = "enabled", havingValue = "true", matchIfMissing = true)
+public class TaskEventAutoConfiguration {
+
+	@Configuration
+	@EnableBinding(TaskEventChannels.class)
```

Comment:
I think it is better to specify the `contentType` of the `task-events` channel to `application/json` here. That way, we don't have to worry about the (both SCSt and non-SCSt) consumer requiring `TaskExecution` in its classpath or an appropriate converter in place.